### ISSUE TITLE
Proposal to add otel.collector.receiver.name attribute

### ISSUE THREAD

**Issue by avanish-vaghela** [open-telemetry/semantic-conventions#759]
**What are you trying to achieve?**
In large organisations telemetry pipeline tends to ingest data from a variety of sources, this in effect makes it harder to debug issues such as missing telemetry data or inconsistent attributes. To facilitate the debugging process, it is crucial that we can easily identify where the telemetry data is emitted from and what receiver is responsible for capturing the telemetry data in question. Currently, there is no native attribute available for tagging the telemetry data captured by a specific receiver and as `otel.*` attributes are reserved for OpenTelemery, one of the workarounds is to create a custom label `<company_name>.otel.collector.receiver.name`.  

**What did you expect to see?**
Ability to set the the value of `otel.collector.receiver.name` attribute.

**Additional Information**
I had raised this question in a number of different forums i.e. CNCF OpenTelemetry Slack channels ([here](https://cloud-native.slack.com/archives/C01N6P7KR6W/p1707475927986919) and [here](https://cloud-native.slack.com/archives/C041APFBYQP/p1708081531062549)) as well as the OTel End User working group. It had become apparent that it's a common use case and the community would benefit from having an official attribute. 
Happy to raise a PR :)

**Comment by danielgblanco** [comment]
After this is discussed, it's probably worth raising an issue in the Collector repo as well. Ideally, it should be receivers adding this attribute before the fan-out occurs in setups where receivers are attached to multiple pipelines. Otherwise, according to the [data ownership logic](https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/README.md#data-ownership), if you were to create an attribute processor solely for the purpose of adding this attribute to telemetry from a given processor, you'd be cloning data (potentially unnecessarily)

**Comment by arminru** [comment]
When looking into this, also the [Instrumentation Scope](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/glossary.md#instrumentation-scope) needs to be taken into consideration.
For non-OTel data picked up by these receivers, there won't be any instrumentation scope present from the start and adding the receiver name (and potentially version) in place of the instrumentation that gathered this telemetry feels like a good fit to me.
Some receivers in the collector already put themselves into the Instrumentation Scope, but not all of them do so (yet?).

**Comment by avanish-vaghela** [comment]
What would be the next steps here?

**Comment by avanish-vaghela** [comment]
@arminru Yes, this makes sense. However, I am not sure how it would translate for a backend like Prometheus. Would the best practice be to convert it to an attribute? If so, would it make sense to have a globally defined resource attribute?

**Comment by jmichalek132** [comment]
Instrumentation scope, afaik is just being dropped in otel collector when translating OTEL metrics to prometheus, but there is an open PR to handle it. https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24248
